sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BeamSchemaToBytesTransformers.java

```diff
@@ -0,0 +1,115 @@
+
+@SuppressWarnings("nullness")
+public class BeamSchemaToBytesTransformers {
+
+  @UnknownInitialization
+  @FunctionalInterface
+  public interface MyBiFunction<T, X, K> extends BiFunction<T, X, K> {
+
+    @Override
+    @Nullable
+    K apply(T t, X u);
+
+    @Override
+    default <V> BiFunction<T, X, V> andThen(Function<? super K, ? extends V> after) {
+      Objects.requireNonNull(after);
+      return (T t, X u) -> after.apply(apply(t, u));
+    }
+  }
+
+  private static byte[] toBytes(@Nullable Object obj) {
+    if (obj == null) {
+      return new byte[0];
+    } else {
+      return obj.toString().getBytes(StandardCharsets.UTF_8);
+    }
+  }
+
+  public static byte[] encodeArrayField(Row row, Schema.Field field) {
+    if (field.getType().getCollectionElementType().equals(Schema.FieldType.DATETIME)) {
+      throw new UnsupportedOperationException("Datetime arrays are not supported!");
+    } else {
+      return Bytes.concat(
+          new byte[] {(byte) '['},
+          row.getArray(field.getName()).stream()
+              .map(obj -> obj == null ? "" : obj.toString())
+              .collect(Collectors.joining(","))
+              .getBytes(StandardCharsets.UTF_8),
+          new byte[] {(byte) ']'});
+    }
+  }
+
+  @SuppressWarnings("DoNotCallSuggester")
+  public static byte[] encodeRowField(Row row, Schema.Field field) {
+    throw new UnsupportedOperationException("Nested fields are not supported!");
+  }
+
+  public static MyBiFunction<Row, Schema.Field, byte[]> getBytesEncoders(Schema.TypeName type) {
+    switch (type) {
+      case BYTES:
+        return ((row, field) -> row.getBytes(field.getName()));
+      case STRING:
+        return ((row, field) -> toBytes(row.getString(field.getName())));
+      case BOOLEAN:
+        return ((row, field) -> toBytes(row.getBoolean(field.getName())));
+      case DOUBLE:
+        return ((row, field) -> toBytes(row.getDouble(field.getName())));
+      case FLOAT:
+        return ((row, field) -> toBytes(row.getFloat(field.getName())));
+      case INT16:
+        return ((row, field) -> toBytes(row.getInt16(field.getName())));
+      case INT32:
+        return ((row, field) -> toBytes(row.getInt32(field.getName())));
+      case INT64:
+        return ((row, field) -> toBytes(row.getInt64(field.getName())));
+      case DECIMAL:
+        return ((row, field) -> toBytes(row.getDecimal(field.getName())));
+      case ARRAY:
+        return BeamSchemaToBytesTransformers::encodeArrayField;
+      case ROW:
+        return BeamSchemaToBytesTransformers::encodeRowField;
+      case DATETIME:
+        return ((row, field) -> {
+          if (row.getDateTime(field.getName()) == null) {
+            return new byte[0];
+          } else {
+            return ISODateTimeFormat.basicDateTime()
+                .print(row.getDateTime(field.getName()))
+                .getBytes(StandardCharsets.UTF_8);
+          }
+        });
+      default:
+        return null;
```

Comment:
why not throw the illegal argument exception here? would make BigTableIOWriteSchemaBasedTransform(lines 239-250) more readable